docs/examples/using-terraform-plugin.md

```diff
@@ -0,0 +1,265 @@
+
+The execution-folder directs the Terraform container to tell it where to execute from as all the terraform files are stored in different directories on this repository.
+
+**NOTE: Terraform container has its own http backend located at 'http://localhost:8001/{state-name}'. If provided args-on-init and state-name we will write the tfstate to a workflow variable.**
+
+### Azure
+The only secrets required to run with Azure is a subscription_id, client_id, client_secret and a tenant_id.
+
+
+```yaml
+  - id: deploy_azure
+    type: action
+    log: .
+    action:
+        secrets: ["CLIENT_ID", "TENANT_ID", "SUBSCRIPTION_ID", "CLIENT_SECRET"]
+        function: tfrun
+        input: 
+            action: "{{.action}}"
+            args-on-init: ["-backend-config=address=http://localhost:8001/terraform-azure-instance"]
+            execution-folder: terraform-examples/azure
+            variables:
+                state-name: terraform-azure-instance
+                subscription_id: "{{.secrets.SUBSCRIPTION_ID}}"
+                client_id: "{{.secrets.CLIENT_ID}}"
+                client_secret: "{{.secrets.CLIENT_SECRET}}"
+                tenant_id: "{{.secrets.TENANT_ID}}"
+    transform: |
+      {
+        "ip": .ip
+      }
+    transition: send_message
+```
+
+### Google Cloud Platform
+The only secrets required to run with Google is a service_account_key and a project_id.
+
+```yaml
+  - id: deploy_gcp
+    type: action
+    log: .
+    action:
+      secrets: ["SERVICE_ACCOUNT_KEY", "PROJECT_ID"]
+      function: tfrun
+      input:
+        execution-folder: terraform-examples/google
+        action: "{{.action}}"
+        args-on-init: ["-backend-config=address=http://localhost:8001/terraform-gcp-instance"]
+        variables:
+          state-name: terraform-gcp-instance
+          project_id: "{{.secrets.PROJECT_ID}}"
+          service_account_key: "{{.secrets.SERVICE_ACCOUNT_KEY}}"
+    transform: |
+      {
+          "ip": .ip
+      }
+    transition: send_message
+```
+
+### Amazon Web Services
+The only secrets required to run with Amazon is an access_key and a secret_key.
+
+```yaml
+  - id: deploy_amazon
+    type: action
+    log: .
+    action:
+      secrets: ["AMAZON_KEY", "AMAZON_SECRET"]
+      function: tfrun
+      input:
+        execution-folder: terraform-examples/amazon
+        action: "{{.action}}"
+        args-on-init: ["-backend-config=address=http://localhost:8001/terraform-amazon-instance"]
+        variables:
+          region: us-east-2
+          state-name: terraform-amazon-instance
+          amazon_key: "{{.secrets.AMAZON_KEY}}"
+          amazon_secret: "{{.secrets.AMAZON_SECRET}}"
+    transition: send_message
+    transform: |
+      {
+          "ip": .ip
+      }
+```
+
+**NOTE: you will find each terraform state has a transform attribute so I can easily get the IP to send via a Discord Message at the end**
```

Comment:
Each terraform state uses the `transform` field to pluck the IP address of the created virtual machine, to be sent to the Discord webhook.